src/main/kotlin/LinkedList.kt

```diff
@@ -10,18 +10,54 @@ class LinkedList<T> {
     
     fun count(): Int = size
        
-    fun insert(index: Int, obj: T) {
-        TODO("Not yet implemented")
+    operator fun get(index: Int): T {
+        var current = head
+        var currentIndex = 0
+        while (current != null && currentIndex < index) {
+            current = current.nextNode
+            currentIndex++
+        }
+        return current
     }
-    fun removeAt(index: Int) {
+    operator fun set(index: Int, obj: T) {
         TODO("Not yet implemented")
     }
-    operator fun get(index: Int): T {
-        TODO("Not yet implemented")
+    
+    fun insertFirst(value: T) {
+        head = Node(value = value, nextNode = head)
+        if (last == null) {
+            last = head
+        }
+        size++
     }
-    operator fun set(index: Int, obj: T) {
+
+    fun add(value: T) {
+        if (size==0) {
+            insertFirst(value)
+        }
+        else {
+            last?.nextNode = Node(value = value)
+            last= last?.nextNode
+            size++
+        }
+    }
+    
+    fun insert(index: Int, value: T) {
+        if (size==0 || index == size) {
+            insertFirst(value)
+        }
```

Comment:
Ничего не понимаю, что здесь происходит. Думаю, такое должно исправить ситуацию.
```suggestion
        require(index in 0 .. size)
        if (index == 0) {
            insertFirst(value)
        }
```

Reply:
ошибся, в if вызов insertLast, где либо insertFirst будет вызван при size==0, либо добавление в конец при index == size